examples/everything/main.go

```diff
@@ -227,6 +236,43 @@ func handleResourceTemplate(
 	}, nil
 }
 
+func handleGeneratedResource(
+	ctx context.Context,
+	request mcp.ReadResourceRequest,
+) ([]mcp.ResourceContents, error) {
+	uri := request.Params.URI
+
+	var resourceNumber string
+	if _, err := fmt.Sscanf(uri, "test://static/resource/%s", &resourceNumber); err != nil {
+		return nil, fmt.Errorf("invalid resource URI format: %w", err)
+	}
+
+	num, err := strconv.Atoi(resourceNumber)
+	if err != nil {
+		return nil, fmt.Errorf("invalid resource number: %w", err)
+	}
+
+	index := num - 1
+
+	if index%2 == 0 {
+		return []mcp.ResourceContents{
+			mcp.TextResourceContents{
+				URI:      uri,
+				MIMEType: "text/plain",
+				Text:     fmt.Sprintf("Text content for resource %d", num),
+			},
+		}, nil
+	} else {
+		return []mcp.ResourceContents{
+			mcp.BlobResourceContents{
+				URI:      uri,
+				MIMEType: "application/octet-stream",
+				Blob:     base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("Binary content for resource %d", num))),
+			},
+		}, nil
+	}
+}
+
```

Comment:
Add a handler to handle both the `text/plain` and `application/octet-stream` resource. It follows the same logic as the `generateResources()` function. Example resource listing:

```json
{
  "contents": [
    {
      "uri": "test://static/resource/77",
      "mimeType": "text/plain",
      "text": "Text content for resource 77"
    }
  ]
}
```

```json
{
  "contents": [
    {
      "uri": "test://static/resource/78",
      "mimeType": "application/octet-stream",
      "blob": "QmluYXJ5IGNvbnRlbnQgZm9yIHJlc291cmNlIDc4"
    }
  ]
}
```